test/e2e/cache/cache_server_test.go

```diff
@@ -0,0 +1,504 @@
+	if !apierrors.IsNotFound(err) {
+		t.Fatalf("expected to get a NotFound error, got %v", err)
+	}
+}
+
+// testSpecStatusSimultaneously checks if updating spec and status at the same time works
+func testSpecStatusSimultaneously(ctx context.Context, t *testing.T, cacheClientRT *rest.Config, cluster logicalcluster.Name, gvr schema.GroupVersionResource) {
+	cacheDynamicClient, err := dynamic.NewClusterForConfig(cacheClientRT)
+	require.NoError(t, err)
+	initialCucumberDB := newFakeAPIExport("cucumberdb")
+
+	t.Logf("Create abmer|%s/%s (shard|cluster/name) on the cache server", cluster, initialCucumberDB.Name)
+	cucumberDBRaw, err := toUnstructured(&initialCucumberDB)
+	require.NoError(t, err)
+	cachedCucumberDBRaw, err := cacheDynamicClient.Cluster(cluster).Resource(gvr).Create(ctx, cucumberDBRaw, metav1.CreateOptions{})
+	require.NoError(t, err)
+	cachedCucumberDBJson, err := cachedCucumberDBRaw.MarshalJSON()
+	require.NoError(t, err)
+	cachedCucumberDB := &fakeAPIExport{}
+	require.NoError(t, json.Unmarshal(cachedCucumberDBJson, cachedCucumberDB))
+
+	t.Logf("Update amber|%s/%s (shard|cluster/name) on the cache server", cluster, initialCucumberDB.Name)
+	cachedCucumberDB.Status.Condition = "run out"
+	cachedCucumberDB.Spec.Size = 1111
+	cachedCucumberDBRaw, err = toUnstructured(cachedCucumberDB)
+	require.NoError(t, err)
+	cachedCucumberDBRaw, err = cacheDynamicClient.Cluster(cluster).Resource(gvr).Update(ctx, cachedCucumberDBRaw, metav1.UpdateOptions{})
+	require.NoError(t, err)
+	cachedCucumberDBJson, err = cachedCucumberDBRaw.MarshalJSON()
+	require.NoError(t, err)
+	cachedCucumberDB = &fakeAPIExport{}
+	require.NoError(t, json.Unmarshal(cachedCucumberDBJson, cachedCucumberDB))
+	if cachedCucumberDB.Spec.Size != 1111 {
+		t.Fatalf("unexpected spec.size %v after an update of amber|%s/%s (shard|cluster/name), epxected %v", cachedCucumberDB.Spec.Size, cluster, initialCucumberDB.Name, 1111)
+	}
+	if cachedCucumberDB.Status.Condition != "run out" {
+		t.Fatalf("unexpected status.condition %v after an update of amber|%s/%s (shard|cluster/name), epxected %v", cachedCucumberDB.Status.Condition, cluster, initialCucumberDB.Name, "run out")
+	}
+
+}
+
+func newFakeAPIExport(name string) fakeAPIExport {
+	return fakeAPIExport{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "apis.kcp.dev/v1alpha1",
+			Kind:       "APIExport",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        name,
+			Annotations: map[string]string{},
+		},
+	}
+}
+
+func toUnstructured(obj interface{}) (*unstructured.Unstructured, error) {
+	unstructured := &unstructured.Unstructured{Object: map[string]interface{}{}}
+	raw, err := runtime.DefaultUnstructuredConverter.ToUnstructured(obj)
+	if err != nil {
+		return nil, err
+	}
+	unstructured.Object = raw
+	return unstructured, nil
+}
+
+// TODO: refactor with TestAllScenariosAgainstStandaloneCacheServer
+func TestCacheServerAllScenarios(t *testing.T) {
```

Comment:
You need a `t.Parallel()` and `framework.Suite()` here

Reply:
done